tests/test_class.cpp

```diff
@@ -47,10 +47,25 @@ TEST_SUBMODULE(class_, m) {
         }
         ~NoConstructor() { print_destroyed(this); }
     };
+    struct NoConstructorNew {
+        NoConstructorNew() = default;
+        NoConstructorNew(const NoConstructorNew &) = default;
+        NoConstructorNew(NoConstructorNew &&) = default;
+        static NoConstructorNew *new_instance() {
+            auto *ptr = new NoConstructorNew();
+            print_created(ptr, "via new_instance");
+            return ptr;
+        }
+        ~NoConstructorNew() { print_destroyed(this); }
+    };
 
     py::class_<NoConstructor>(m, "NoConstructor")
         .def_static("new_instance", &NoConstructor::new_instance, "Return an instance");
 
+    py::class_<NoConstructorNew>(m, "NoConstructorNew")
+        .def(py::init([](NoConstructorNew *self) { return self; })) // Need a NOOP __init__
```

Comment:
I also wish there was a cleaner way to not have to implement `__init__` if __new__ is defined, but I can't figure out how to hack to the metaclass not to complain.

Reply:
Maybe something like `.def_new(...)` or `.def(py::new(...))` that wraps the two methods you have right now?